backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/product/ProductRestService.java

```diff
@@ -377,6 +383,12 @@ else if (effectiveSyncAdvise.isFailIfNotExists())
 		else
 		{
 			final CreateBPartnerProductRequest createBPartnerProductRequest = getCreateBPartnerProductRequest(jsonRequestBPartnerProductUpsert, productId, bPartnerId);
+
+			if (Boolean.TRUE.equals(createBPartnerProductRequest.getCurrentVendor()))
+			{
+				productRepository.resetCurrentVendorFor(productId);
```

Comment:
..same as above

Reply:
this is on create flow. we validate on `createBPartnerProductRequest` as there is no existing record